In [1]:
#Import Libraries
import pandas as pd
import numpy as np
import scipy.stats as stats

In [2]:
#Create DataFrame
transactions = [
    100, 150, 50, 100, 130, 120, 100, 85, 70, 150,
    150, 120, 50, 100, 100, 140, 90, 150, 50, 90,
    120, 100, 110, 75, 65
]
df = pd.DataFrame(transactions, columns=['TransactionAmount'])

In [4]:
#Central Tendency
mean_value = df['TransactionAmount'].mean()
median_value = df['TransactionAmount'].median()
mode_value = df['TransactionAmount'].mode()[0]

print("Mean:", mean_value)
print("Median:", median_value)
print("Mode:", mode_value)

Mean: 102.6
Median: 100.0
Mode: 100


Rata-rata (Mean) transaksi: $102.6

Median transaksi: $100

Mode transaksi: $100

Insight: Mayoritas transaksi tetap dekat dengan rata-rata sebelumnya $100.

In [5]:
#Variability
variance = df['TransactionAmount'].var()
std_dev = df['TransactionAmount'].std()
max_val = df['TransactionAmount'].max()
min_val = df['TransactionAmount'].min()
q1 = df['TransactionAmount'].quantile(0.25)
q2 = df['TransactionAmount'].quantile(0.5)  # ini sama dengan median
q3 = df['TransactionAmount'].quantile(0.75)

print("Variance:", variance)
print("Standard Deviation:", std_dev)
print("Max:", max_val, "Min:", min_val)
print("Q1 (25%):", q1)
print("Q2 (50%/Median):", q2)
print("Q3 (75%):", q3)

Variance: 1012.7499999999997
Standard Deviation: 31.82373328193912
Max: 150 Min: 50
Q1 (25%): 85.0
Q2 (50%/Median): 100.0
Q3 (75%): 120.0


-Meskipun rata-rata dekat $100, beberapa transaksi jauh dari rata-rata (ada variasi tinggi)

-Dari hasil Min dan Max, ada beberapa transaksi sangat tinggi atau sangat rendah dibandingkan rata-rata

-Ada beberapa outlier di bawah 85 dan di atas 120

In [ ]:
#Define H0 and H1
#H0: Rata-rata transaksi = $100 (tidak berubah)
#H1:Rata-rata transaksi ≠ $100 (berubah)

In [6]:
#Define alpha = 0.05
alpha = 0.05

In [7]:
#Calculate t statistics and P-value
#Perform one sample t-test

population_mean = 100

# T-Test (one-sample)
t_stat, p_value = stats.ttest_1samp(df['TransactionAmount'], population_mean)

...

Ellipsis

In [8]:
print("T-Statistic:", t_stat)
print("P-Value:", p_value)

T-Statistic: 0.4085001556802841
P-Value: 0.6865284813438117


In [10]:
#Conclude with alpha and p-value
if p_value < alpha:
    print("H0 ditolak → Rata-rata transaksi berubah setelah training")
else:
    print("H0 tidak ditolak → Tidak ada bukti rata-rata transaksi berubah")
#Conclusion: H0 tidak ditolak → Tidak ada bukti rata-rata transaksi berubah

H0 tidak ditolak → Tidak ada bukti rata-rata transaksi berubah


In [12]:
#Conclude with t statistics and critical region
#Find the critical region
n = len(df)
dfree = n - 1
t_critical_left = stats.t.ppf(alpha/2, dfree)
t_critical_right = stats.t.ppf(1 - alpha/2, dfree)

In [ ]:
#H1: berlaku jika T-Statistic berada di luar critical region

In [14]:
#Critical region result
print("Critical Region: T < {:.3f} or T > {:.3f}".format(t_critical_left, t_critical_right))
#Conclusion: T-Statistic 0.41 berada di dalam batas critical region (-2.06 sampai 2.06), Artinya tidak cukup bukti untuk menolak H0
#rata-rata transaksi setelah training tidak berbeda secara signifikan dari $100

Critical Region: T < -2.064 or T > 2.064
